### PR TITLE
fix(sagaInjectors) Prevent sagas to be injected multiple times

### PR DESCRIPTION
**Pull request by lvarayut**
This PR prevents the sagas to be injected if its key is duplicate with the key inside the `injectedSagas` object.

When a duplicate key passed to the `injectSaga()` function, the current saga in the `injectedSagas` object will be replaced with the newly created one. So, there is no way we can refer to the old saga and cancel it. Therefore, we need to be careful to not inject a saga that has a duplicate "key". 

Related to #1983 and #2061. 

Before opening a pull request, please ensure:

- [x] You have followed our [**contributing guidelines**](https://github.com/react-boilerplate/react-boilerplate/blob/master/.github/CONTRIBUTING.md)
- [x] double-check your branch is based on `dev` and targets `dev` 
- [x] Pull request has tests (we are going for 100% coverage!)
- [x] Code is well-commented, linted and follows project conventions
- [x] Documentation is updated (if necessary)
- [x] Internal code generators and templates are updated (if necessary)
- [x] Description explains the issue/use-case resolved and auto-closes related issues

### CODE REVIEW

**Comment by mxstbr** [comment]
Deploy preview failed.

Built with commit a1eabe0b9842fe26f732a6cd0e9cd54e112e56b2

https://app.netlify.com/sites/react-boilerplate/deploys/5a315f45a1147764782632d0

File: app/utils/sagaInjectors.js [hunk 1]
```diff
@@ -29,6 +29,15 @@ const checkDescriptor = (descriptor) => {
   );
 };
 
+const checkRunningSaga = (store, key) => {
+  const descriptor = store.injectedSagas[key];
+  const isRunning = descriptor && descriptor.task && descriptor.task.isRunning();
+  invariant(
+    !isRunning,
```
Comment (blling) [inline app/utils/sagaInjectors.js]: If developers accidentally introduced the same key saga So the original saga will be covered, I think we should not be allowed to pass the same key saga

**Comment by blling** [comment]
cc @Dattaya

File: app/utils/sagaInjectors.js (hunk 1 above)
Comment (lvarayut) [reply app/utils/sagaInjectors.js]: If the saga has already cancelled, the developers should be allowed to pass the same key to start the saga again. Therefore, `checkRunningSaga` should check only if the saga is duplicate and also running.

**Pull request closed without merging**